File: server/src/main/java/io/github/mufasa1976/calcmaster/records/MultiplicationProperties.java
package io.github.mufasa1976.calcmaster.records;

import io.github.mufasa1976.calcmaster.enums.MultiplicationType;
import lombok.Builder;

import java.util.List;

@Builder
public record MultiplicationProperties(int maxProduct, List<Integer> exclusions, List<Integer> fixedMultiplicands, int transgression, MultiplicationType type) {}

File: frontend/src/shared/multiplication-properties.ts
import { UNLIMITED_TRANSACTIONS } from "./constants";

export enum MultiplicationType {
  SQRT = "SQRT",
  HALF_OF_PRODUCT = "HALF_OF_PRODUCT"
}

export interface MultiplicationProperties {
  maxProduct: number;
  exclusions: number[];
  fixedMultiplicands: number[];
  transgression: number;
  type: MultiplicationType;
}

export const initialMultiplicationProperties = {
  maxProduct: 100,
  exclusions: [],
  fixedMultiplicands: [],
  transgression: UNLIMITED_TRANSACTIONS,
  type: MultiplicationType.SQRT
} as MultiplicationProperties;
